src/pop/c/entities/btcblock.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2019-2021 Xenios SEZC
+// https://www.veriblock.org
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+#include <memory>
+
+#include "veriblock/pop/c/entities/btcblock.h"
+#include "veriblock/pop/entities/btcblock.hpp"
+
+POP_ENTITY_STRUCT_NAME(btc_block) {
+  std::shared_ptr<altintegration::BtcBlock> ref;
+};
+
+POP_ENTITY_NEW_SIGNATURE(btc_block) { return new POP_ENTITY_NAME(btc_block)(); }
+
+POP_ENTITY_FREE_SIGNATURE(btc_block) {
+  if (self != nullptr) {
+    delete self;
+    self = nullptr;
+  }
+}
+
+POP_ENTITY_GETTER_SIGNATURE(btc_block, POP_ARRAY_NAME(u8), hash) {
+  auto hash = self->ref->getHash();
```

Comment:
`getHash` returns `const uint256&` which means that this array can point to this hash as slice

Reply:
I think it would make sense to have separate `array` and `slice` types. Where `array` would own memory, and `slice` will be a pointer.